### DISCUSSION TITLE
Model Architecture

### DISCUSSION THREAD

**Discussion by rkoystart** [explosion/spaCy#11356] [Help: Other Questions] [answered]
Hi is there a way to view the layers present in a spacy model (en_core_web_sm). So my question is,  In pytorch the following statement
`
print(model)
`
will print the architecture (layers within the model), which gives us an easy undertanding on how the input is being transformed or operated.

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @rkoystart , you can use something like `nlp.pipe_names` to see the components that underlie a particular spaCy model. You can couple this with `nlp.get_pipe_config` to see how that particular component was configured. For example:

```python
import spacy

nlp = spacy.load("en_core_web_lg")
for pipe in nlp.pipe_names:
    print(nlp.get_pipe_config(pipe))
```

You can then cross-reference them with the architectures in the [spaCy documentation](https://spacy.io/api/architectures). Note that a spaCy pipeline will not always contain only neural models, so this method is not exactly a one-to-one comparison with pytorch (that gives you the layers).
